dart/tool/dart/generate_dart.dart

```diff
@@ -620,37 +620,39 @@ abstract class VmServiceInterface {
         Response response;
 
         switch(method) {
-          case '_registerService':
+          case 'registerService':
             $_registerServiceImpl
             break;
     ''');
     methods.where((m) => !m.isUndocumented).forEach((m) {
-      gen.writeln("case '${m.name}':");
-      if (m.name == 'streamListen') {
-        gen.writeln(_streamListenCaseImpl);
-      } else if (m.name == 'streamCancel') {
-        gen.writeln(_streamCancelCaseImpl);
-      } else {
-        gen.write("response = await _serviceImplementation.${m.name}(");
-        // Positional args
-        m.args.where((arg) => !arg.optional).forEach((arg) {
-          gen.write("params['${arg.name}'], ");
-        });
-        // Optional named args
-        var namedArgs = m.args.where((arg) => arg.optional);
-        if (namedArgs.isNotEmpty) {
-          namedArgs.forEach((arg) {
-            if (arg.name == 'scope') {
-              gen.writeln(
-                  "${arg.name}: params['${arg.name}']?.cast<String, String>(), ");
-            } else {
-              gen.writeln("${arg.name}: params['${arg.name}'], ");
-            }
+      if (m.name != 'registerService') {
+        gen.writeln("case '${m.name}':");
+        if (m.name == 'streamListen') {
+          gen.writeln(_streamListenCaseImpl);
+        } else if (m.name == 'streamCancel') {
+          gen.writeln(_streamCancelCaseImpl);
+        } else {
+          gen.write("response = await _serviceImplementation.${m.name}(");
```

Comment:
Adding @jakemac53, who added the code to handling delegating to registered extensions (https://github.com/dart-lang/vm_service_drivers/commit/d201c622b732d9b495cceb43fd2e8a65c925bd6f).

Jake, `_registerService ` is being renamed to `registerService `.

Reply:
This looks correct, we special case `registerService`  for the server implementation as we have a generic implementation via the ServiceExtensionRegistry.

Reply:
It might make sense code organization wise to move the code on line 623  here though:

```dart
} else if (m.name == 'registerService') {
 gen.writeln(_registerServiceImpl);
}
```

Reply:
Good call. Done.